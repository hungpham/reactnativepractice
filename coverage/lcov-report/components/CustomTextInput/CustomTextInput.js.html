<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/CustomTextInput/CustomTextInput.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/CustomTextInput</a> CustomTextInput.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* @flow */
&nbsp;
import React, { Component, PropTypes } from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  Image,
  Platform
} from 'react-native';
&nbsp;
import applicationStyles from 'config/applicationStyle';
import Metrics from 'config/metrics';
import Colors from 'config/colors';
&nbsp;
export default class CustomTextInput extends Component {
  render() {
    return (
      &lt;View style={styles.inputWrap}&gt;
        &lt;View style={styles.iconWrap}&gt;
          &lt;Image
            source={this.props.imageIcon}
            style={applicationStyles.iconInputField} /&gt;
        &lt;/View&gt;
        &lt;TextInput
          {...this.props}
          style={applicationStyles.inputField}
          underlineColorAndroid={Colors.transparent}
          placeholderTextColor={this.props.placeholderTextColor ? this.props.placeholderTextColor: Colors.Alto}
        /&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
CustomTextInput.propTypes = {
  imageIcon: PropTypes.number.isRequired
};
&nbsp;
const styles = StyleSheet.create({
  inputWrap: {
    paddingLeft: Metrics.paddingLeft,
    height: Metrics.inputFieldHeight,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    // backgroundColor: 'green',
    borderBottomWidth: 0.5,
    borderBottomColor: "#CCC",
  },
  iconWrap: {
    padding: Metrics.padding,
    marginLeft: 2*Metrics.marginHorizontal,
    // backgroundColor: 'red'
  }
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Mar 01 2017 17:39:14 GMT+0700 (ICT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
