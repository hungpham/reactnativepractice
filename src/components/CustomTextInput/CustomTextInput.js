/* @flow */

import React, { Component, PropTypes } from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  Image,
  Platform
} from 'react-native';

import applicationStyles from 'config/applicationStyle';
import Metrics from 'config/metrics';
import Colors from 'config/colors';

export default class CustomTextInput extends Component {
  render() {
    return (
      <View style={styles.inputWrap}>
        <View style={styles.iconWrap}>
          <Image
            source={this.props.imageIcon}
            style={applicationStyles.iconInputField} />
        </View>
        <TextInput
          {...this.props}
          style={applicationStyles.inputField}
          underlineColorAndroid={Colors.transparent}
          placeholderTextColor={this.props.placeholderTextColor ? this.props.placeholderTextColor: Colors.Alto}
        />
      </View>
    );
  }
}

CustomTextInput.propTypes = {
  imageIcon: PropTypes.number.isRequired
};

const styles = StyleSheet.create({
  inputWrap: {
    paddingLeft: Metrics.paddingLeft,
    height: Metrics.inputFieldHeight,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    // backgroundColor: 'green',
    borderBottomWidth: 0.5,
    borderBottomColor: "#CCC",
  },
  iconWrap: {
    padding: Metrics.padding,
    marginLeft: 2*Metrics.marginHorizontal,
    // backgroundColor: 'red'
  }
});
