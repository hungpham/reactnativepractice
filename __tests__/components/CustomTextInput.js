import React from 'react';
import renderer from 'react-test-renderer';
import CustomTextInput from 'components/CustomTextInput/CustomTextInput';

it('renders correctly', () => {
  const tree = renderer.create(
		<CustomTextInput
			autoCapitalize={'none'}
			onChangeText={(text) => this.setState({username: text})}
			keyboardType={'email-address'}
			placeholder={'UserName'}
			imageIcon={require('assets/images/user_name.png')} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
