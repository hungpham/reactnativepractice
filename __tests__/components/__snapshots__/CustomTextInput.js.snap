// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#CCC",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingLeft": undefined,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 20,
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "resizeMode": "contain",
          "width": 20,
        }
      }
    />
  </View>
  <TextInput
    autoCapitalize="none"
    imageIcon={1}
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="UserName"
    placeholderTextColor="#d7d7d7"
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "color": "white",
        "flex": 1,
        "paddingHorizontal": 10,
      }
    }
    underlineColorAndroid="rgba(0, 0, 0, 0)"
  />
</View>
`;
